components/QueryInfo/QueryInfo.tsx

```diff
@@ -0,0 +1,60 @@
+import { CheckCircleIcon, AlertFillIcon } from '@primer/octicons-react'
+import React from 'react'
+import { Spinner } from 'react-bootstrap'
+import styles from '../../scss/queryInfo.module.scss'
+
+type QueryInfo<T> = {
+  data: T
+  loading: boolean
+  error: string
+  texts: {
+    loading: string
+    errorTitle: string
+  }
+  DataMessage?: () => JSX.Element
+}
+
+const QueryInfo = <T,>({
+  data,
+  loading,
+  error,
+  texts,
+  DataMessage
+}: QueryInfo<T>) => {
+  if (loading) {
+    return (
+      <div className={styles.loading}>
+        <Spinner animation="grow" size="sm" />
+        <span>{texts.loading}</span>
+      </div>
+    )
+  }
+
+  if (error) {
+    return (
+      <div className={styles.error}>
+        <AlertFillIcon />
+        <span>
+          {texts.errorTitle}: {error}
```

Comment:
This component seems like it has high potential to be used in a public-facing feature in the future. In that case, we shouldn't display the actual error coming from the request, that would be bad user experience for the user to see error logs that are for the developers to see